python/vyos/ifconfig/ethernet.py

```diff
@@ -417,12 +418,38 @@ def set_ring_buffer(self, rx_tx, size):
             print(f'could not set "{rx_tx}" ring-buffer for {ifname}')
         return output
 
+    def set_switchdev(self, enable):
+        ifname = self.config['ifname']
+        if not bool(re.match(r"^eth[0-9]+$", ifname)):
+            print(f'switchdev mode is only possible on physical interfaces!')
+            return
+
+        addr = self.ethtool.get_bus_info()
+        if addr == None:
+            print(f'could not resolve PCIe address of {ifname}')
+            return
+
+        enabled = False
+        state, code = self._popen(f'devlink dev eswitch show pci/{addr} | awk \'{{print $3}}\'')
+        if code == 0 and state == 'switchdev':
+            enabled = True
+
+        if enable and not enabled:
+            output, code = self._popen(f'/sbin/devlink dev eswitch set pci/{addr} mode switchdev')
```

Comment:
Why now with an absolute path and in line 433 no absolute path?

Reply:
fixed